src/main/java/cn/edu/thu/tsfile/common/conf/TSFileDescriptor.java

```diff
@@ -60,22 +60,66 @@ private void loadYaml() {
 				System.exit(1);
 			}
 		}
-		LOGGER.info("start to read config file {}", url);
+		LOGGER.info("Start to read config file {}", url);
+		Properties properties = new Properties();
 		try {
-			Yaml yaml = new Yaml();
-			conf = yaml.loadAs(inputStream, TSFileConfig.class);
-		} catch (Exception e) {
-			LOGGER.error("Loading settings {} failed.", url, e);
-			System.exit(1);
-		} finally {
-			if (inputStream != null) {
-				try {
-					inputStream.close();
-					inputStream = null;
-				} catch (IOException e) {
-				}
-			}
+		    properties.load(inputStream);
+		    
+		    String rowGroupSize = properties.getProperty("rowGroupSize");
+		    conf.rowGroupSize = Integer.parseInt(rowGroupSize);
+		    
+		    String pageSize = properties.getProperty("pageSize");
+		    conf.pageSize = Integer.parseInt(pageSize);
+
+		    String timeSeriesEncoder = properties.getProperty("timeSeriesEncoder");
+		    conf.timeSeriesEncoder = timeSeriesEncoder;		    
+		    
+		    String defaultSeriesEncoder = properties.getProperty("defaultSeriesEncoder");
+		    conf.defaultSeriesEncoder = defaultSeriesEncoder;		    
+		    
+		    String compressName = properties.getProperty("compressName");
+		    conf.compressName = compressName;		    
+		    
+		    String defaultRleBitWidth = properties.getProperty("defaultRleBitWidth");
+		    conf.defaultRleBitWidth = Integer.parseInt(defaultRleBitWidth);
+
+		    String defaultEndian = properties.getProperty("defaultEndian");
+		    conf.defaultEndian = defaultEndian;
+		    
+		    String defaultDeltaBlockSize = properties.getProperty("defaultDeltaBlockSize");
+		    conf.defaultDeltaBlockSize = Integer.parseInt(defaultDeltaBlockSize);
+		    
+		    String defaultPLAMaxError = properties.getProperty("defaultPLAMaxError");
+		    conf.defaultPLAMaxError = Double.parseDouble(defaultPLAMaxError);
+		    
+		    String defaultSDTMaxError = properties.getProperty("defaultSDTMaxError");
+		    conf.defaultSDTMaxError = Double.parseDouble(defaultSDTMaxError);
+		} catch (IOException e) {
+		    LOGGER.warn("Error format in config file, use default configuration", e);

```

Comment:
Notice that: `System.getproperty()` method returns `the string value of the system property, or null if there is no property with that key.`.
That is to say,  if the properties has no desired property, your conf.field is null!